jetty-ee10/jetty-ee10-maven-plugin/src/main/java/org/eclipse/jetty/ee10/maven/plugin/Overlay.java

```diff
@@ -89,6 +89,8 @@ public void unpackTo(File dir) throws IOException // TODO: change to Path
         // only unpack if the overlay is newer
         if (!Files.exists(pathDir))
         {
+            // create directory
+            Files.createDirectories(pathDir);
```

Comment:
HHmmm, this seems like a change to the previous behaviour of `Resource.copyTo`, specifically that currently the dir will be created during the copy, but now it must pre-exist.

Reply:
The destination in `copyTo(destination)` is how the behavior is determined.

If the source doesn't exist ...
 * always an error

If the source is directory - `srcDir.copyTo(...)`
  * if the destination is a file that exists, that's an error from copyTo.
  * if the destination is a directory that exists, use it to copy directory to directory
  * if the destination doesn't exist, then what do we do (file or dir?), right now we error.  This was one of the questions posed in #11539

If the source is a file - `srcFile.copyTo(...)`
  * if the destination is a file that exists, we copy contents to file (with truncate).
  * if the destination is a directory that exists, we copy the file to the destination directory using the same filename.
  * if the destination doesn't exist, then we only open the file (eg: `newInputStream(CREATE, WRITE)`) and write. (if the parent directories also do not exists, then that's an IOException from Java)

_Scenario 1:_
src: `/path/to/webapp/` (exists)
dest: `/tmp/jetty/webapp-123` (doesn't exists, only the `/tmp/jetty` dir in that path does exist)

This would be a need to create only 1 directory (the `webapp123` path at the end), which is in line with how srcDir.copyTo (point 2) works.

_Scenario 2:_
src: `/path/to/webapp/` (exists)
dest: `/tmp/jetty/webapp-123` (doesn't exists, only the `/tmp` dir in that path does exist)

This would be a need to create only 2 directories (the `jetty/webapp123` paths), which is actually more in line with how srcDir.copyTo (point 3) behaves.

Should we adopt for copyTo(dirPath) point 3 to only allow for Files.createDirectory() and not Files.createDirectories(), just to make the two side equivalent?   If so, then we still need the above changes in Overlay as `pathDir` isn't 1 deep of missing directories.

Reply:
I updated the API documentation in Resource.copyTo(Path), please check the new documentation.

Reply:
If the destination is a directory and it did not exist, clearly we did not throw an error and `Resource.copyTo` created it, otherwise all of the maven plugin tests would have been failing. So I suggest we retain that behaviour.

Reply:
"If the destination is a directory and it did not exist,"

That is an impossible scenario.
If it doesn't exist, it's not a directory (or a file, or a symlink, etc)

Reply:
https://github.com/jetty/jetty.project/pull/11540/files#r1537491034

So how did the maven integration tests work before this change to pre-create the directory? Something in `Resource.copyTo()` must have been handling this scenario?